buildSrc/src/main/java/org/elasticsearch/gradle/internal/InternalDistributionArchiveCheckPlugin.java

```diff
@@ -0,0 +1,311 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle.internal;
+
+import org.elasticsearch.gradle.LoggedExec;
+import org.elasticsearch.gradle.VersionProperties;
+import org.gradle.api.Action;
+import org.gradle.api.GradleException;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.plugins.BasePlugin;
+import org.gradle.api.specs.Spec;
+import org.gradle.api.tasks.TaskProvider;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.stream.Collectors;
+
+public class InternalDistributionArchiveCheckPlugin implements Plugin<Project> {
+
+    @Override
+    public void apply(Project project) {
+        project.getPlugins().apply(BasePlugin.class);
+        String buildTaskName = calculateBuildTask(project.getName());
+        Task buildDistTask = project.getParent().getTasks().getByName(buildTaskName);
+        DistributionArchiveCheckExtension distributionArchiveCheckExtension = project.getExtensions()
+            .create("distributionArchiveCheck", DistributionArchiveCheckExtension.class);
+
+        File archiveExtractionDir = calculateArchiveExtractionDir(project);
+        Spec<Task> toolAvailableSpec = task -> toolExists(project);
```

Comment:
How much effort would it be to rework these implementations to not depend on the CLI tools and instead use Gradle provided archive operation APIs? As it if essentially if our logic to find `tar` or `zip` on the classpath breaks, we'd just start silently skipping these checks.

Reply:
wasn't too much effort and made the plugin way simpler. 👍